File: apps/server/api/src/routes/usage.ts
/**
 * Usage Quota Routes (/api/usage, /api/providers/:id/usage).
 */
import { Hono } from "hono";
import { UsageService } from "@/api/src/services/usage.service.js";
import type { ProviderId } from "@console/types";

export const usageRoutes = new Hono();
const usageService = new UsageService();

/**
 * GET /api/usage — Aggregated usage for all quota-backed providers.
 */
usageRoutes.get("/usage", async (c) => {
  const data = await usageService.getAllUsage(c.req.raw.signal);
  return c.json({ success: true, data });
});

/**
 * GET /api/providers/:id/usage — Usage for a single provider.
 */
usageRoutes.get("/providers/:id/usage", async (c) => {
  const providerId = c.req.param("id") as ProviderId;
  if (providerId !== "gemini" && providerId !== "antigravity" && providerId !== "codex") {
    return c.json({ success: false, error: `Invalid provider '${providerId}' for usage.` }, 400);
  }

  const report = await usageService.getUsage(providerId, c.req.raw.signal);
  return c.json({
    success: true,
    data: report,
  });
});

File: apps/server-go/internal/routes/usage.go
// Usage routes (/api/usage, /api/providers/:id/usage). Port of
// apps/server/api/src/routes/usage.ts.
package routes

import (
	"github.com/gofiber/fiber/v2"

	"github.com/Adelodunpeter25/console/apps/server-go/internal/providers/usage"
)

func registerUsageRoutes(app *fiber.App, svc *usage.Service) {
	app.Get("/api/usage", func(c *fiber.Ctx) error {
		return c.JSON(fiber.Map{"success": true, "data": svc.GetAllUsage(c.Context())})
	})

	app.Get("/api/providers/:id/usage", func(c *fiber.Ctx) error {
		report, err := svc.GetUsage(c.Context(), c.Params("id"))
		if err != nil {
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"success": false, "error": err.Error()})
		}
		return c.JSON(fiber.Map{"success": true, "data": report})
	})
}
